src/tools/crawler/crawler.rs

```diff
@@ -36,21 +37,27 @@ pub(super) fn crawl(
     // Wrapped in box to allow for async recursion.
     async move {
         tokio::spawn(async move {
-            trace!("Crawling {}", addr);
-            if known_network.new_node(addr).await {
-                loop {
-                    tokio::spawn(try_handshake(addr, known_network.clone()));
-                    let success = try_crawling(client.clone(), addr, known_network.clone()).await;
-                    if !success {
-                        let failures = known_network.increase_connection_failures(addr).await;
-                        if failures == u8::MAX {
-                            warn!("Giving up connecting to {}", addr);
-                            break;
-                        }
+            if !known_network.new_node(addr).await {
+                trace!("Skip crawling a known node {addr}");
+                return;
+            }
+
+            trace!("Crawling {addr}");
+            loop {
+                tokio::spawn(try_handshake(addr, known_network.clone()));
+
+                let success = try_crawling(client.clone(), addr, known_network.clone()).await;
+                if !success {
+                    let failures = known_network.increase_connection_failures(addr).await;
+                    if failures == u8::MAX {
+                        warn!("Giving up connecting to {addr}");
+                        break;
                     }
-                    // Even if connection was successful try again after a while to update peers.
-                    sleep(DELAY_AFTER_CONNECTION).await;
                 }
+
+                // Even if connection was successful - try again after a while to update peers.
+                let duration = rand::thread_rng().gen_range(5..=DELAY_AFTER_CONNECTION);
```

Comment:
every 5 seconds seems pretty often, doesn't it :thinking:? I wouldn't reconnect more often than once every 3 minutes

Reply:
With the old solution, where we tried to reconnect every 5 minutes (a fixed time), the number of good nodes took up to 30 minutes to reach number 50 (it doesn't go beyond that - maybe it's because of my VPN)

while with this solution, it reaches that number within 5 minutes

Reply:
but how I see it, the first few iterations where we scout the network should be more frequent,
then once we crawled most of the nodes multiple times, then we can slow down our crawling retries.

That would be the best approach to get reasonable results quickly and allow the crawler to run for a long time while it won't spam the network too much

Reply:
but I don't want to increase the complexity here. but I do like the part where we randomize the request time periods.
is it okay if we configure it to send requests every 3-5 minutes?

for my local tests, I can always tweak those values :-)

Reply:
yeah, randomizing is a good idea, I'd just like to avoid any safeguards to trigger in the nodes we connect to :+1:

Reply:
> I'd just like to avoid any safeguards to trigger in the nodes we connect to

hmm, that sounds like an interesting defensive mechanism, didn't occur to me haha
(seems like it doesn't exist here though)

Done (the change)